Reference for stage-3 arraybuffer-base64 (#36387)
mdn/content: files/en-us/web/javascript/reference/global_objects/uint8array/setfromhex/index.md

---
title: Uint8Array.setFromHex()
slug: Web/JavaScript/Reference/Global_Objects/Uint8Array/setFromHex
page-type: javascript-instance-method
browser-compat: javascript.builtins.Uint8Array.setFromHex
---

{{JSRef}}

The **`setFromHex()`** method of {{jsxref("Uint8Array")}} instances populates this `Uint8Array` object with bytes from a hex-encoded string, returning an object indicating how many bytes were read and written.

> This method parses the string into a byte array. To convert the string into a single number, use the {{jsxref("Global_Objects/parseInt", "parseInt()")}} function with `radix` set to `16` instead.

Reviewer: Is this directly relevant. Feels like an aside.

```suggestion
> [!NOTE]
> To convert the string into a single number instead of a byte array, use the {{jsxref("Global_Objects/parseInt", "parseInt()")}} function with `radix` set to `16` instead.
```

Author: In general when we put up these "use X instead" notes in the intro we don't note-ify it because it's not _that_ important. We use notes for caveats etc.

Reviewer: A matter of taste really. I would have as a note in Web API. But your call.

## Syntax

```js-nolint
setFromHex(string)
```

### Parameters

- `string`

>   - : A hexadecimal string to convert to a `Uint8Array`. The string must only contain characters in the hexadecimal alphabet, which includes 0–9 and A–F (case-insensitive). It must have an even number of characters. Unlike {{jsxref("Uint8Array.prototype.setFromBase64()")}}, the input cannot contain whitespace. Note that the string is only read up to the point where the array is filled, so any invalid hex syntax after that point is ignored.

Reviewer: I'm not sure the parallel with setFromBase64 is as important as the fact that the input cannot contain whitespace, so perhaps reorder

```suggestion
  - : A hexadecimal string to convert to a `Uint8Array`. The string must only contain characters in the hexadecimal alphabet, which includes 0–9 and A–F (case-insensitive). It must have an even number of characters, and it cannot contain whitespace (unlike {{jsxref("Uint8Array.prototype.setFromBase64()")}}). Note that the string is only read up to the point where the array is filled, so any invalid hex syntax after that point is ignored.
```

### Return value

An object containing the following properties:

- `read`

>   - : The number of hex characters read from the input string. It is either the length of the input string, if the decoded data fits into the array, or a multiple of 2 up to the last 2-character chunk that fits into the array.

Reviewer: "or a multiple of 2 up to the last 2-character chunk that fits into the array." is correct but confusing

```suggestion
  - : The number of hex characters read from the input string. If the decoded data fits into the array it is the length of the input string: otherwise it is a the number of complete hex characters that tit into the array.
```
